src/Modules/EngineDemoRecorder.cpp

```diff
@@ -390,10 +391,25 @@ void EngineDemoRecorder::RecordData(const void *data, unsigned long length) {
 }
 
 ON_EVENT(PRE_TICK) {
-	if (event.simulating && !engine->demorecorder->hasNotified && engine->demorecorder->m_bRecording) {
-		const char *cmd = "echo \"SAR " SAR_VERSION " (Built " SAR_BUILT ")\"";
-		engine->SendToCommandBuffer(cmd, 300);
-		engine->demorecorder->hasNotified = true;
+
+	if (engine->demorecorder->m_bRecording) {
+		if (event.simulating && !engine->demorecorder->hasNotified) {
+			const char *cmd = "echo \"SAR " SAR_VERSION " (Built " SAR_BUILT ")\"";
+			engine->SendToCommandBuffer(cmd, 300);
+			engine->demorecorder->hasNotified = true;
+		}
+		for (int i = g_ent_slot_serial.size() - 1; i >= 0; i--) {
```

Comment:
This technically works, but I don't like it. A much clearer pattern is as follows:
```
while (!g_ent_slot_serial.empty()) {
  EntSlotSerial val = g_ent_slot_serial.back();
  if (val.done) {
    // stuff
    g_ent_slot_serial.pop_back();
  }
}
```

Reply:
Actually, you should use `front` and `pop_front`, since you push onto the back of the queue - my bad

Reply:
As you wrote it it would hang. efb0b09 logically uses an iterator for it lol

Reply:
i am very silly as it turns out
the original might honestly be nicer then but i'll sort that myself when i merge
apologies for inconvenience